### PR TITLE
conf: Expose ExecutorsEnabled to frontend if executor.accessToken is set

### PR DESCRIPTION
**Pull request by efritz**
Fixes https://github.com/sourcegraph/sourcegraph/issues/27866. Instead of creating a new flag to enable executors or document that you need to also enable another feature _first_ to get this set up, we should just guard the executor-specific pages when the accessToken is set in the site-config.

Side-effects: We now need to set the `executor.accessToken` in dev our clones of dev-private when working locally as I've removed the fallback environment variable (that happens to be set in sg.conf.yaml for the executor service to read).

### CODE REVIEW

**Review by chrismwendt — APPROVED** [review]
:sparkles:

**Review by eseliger — APPROVED** [review]
I think this is alright for now, but I would actually want to show it all the time later. If none are set up so far it would just be a friendly reminder what you are missing out on and how to fix it.

File: client/web/src/enterprise/site-admin/sidebaritems.ts [hunk 1]
```diff
@@ -32,13 +32,7 @@ const maintenanceGroup: SiteAdminSideBarGroup = {
...
             to: '/site-admin/executors',
             label: 'Executors',
-            // TODO - expand this to executors enabled when SSBC need this page
-            // as well. Right now we don't have an easy way to check if the
-            // executor accessToken is set in site-config, but that should be
-            // the condition of showing this.
-            condition: ({ batchChangesEnabled, batchChangesExecutionEnabled }) =>
-                Boolean(window.context?.codeIntelAutoIndexingEnabled) ||
-                (batchChangesEnabled && batchChangesExecutionEnabled),
+            condition: () => Boolean(window.context?.executorsEnabled),
```
Comment (chrismwendt) [inline client/web/src/enterprise/site-admin/sidebaritems.ts:35]: Drive-by thought: always show "Executors" in the list (for discoverability), and render an info page instead of the list of executors when you haven't set an access token yet.

Comment (efritz) [reply client/web/src/enterprise/site-admin/sidebaritems.ts:35]: I think that's exactly what @eseliger was saying. Not sure who else is working on this right now though and what docs will be available by branch cut. If we want to show links to auto-indexing/batch changes from executors splash page when not enabled we can certainly do that super low-effort (Now that we expose this key).
